tests/test_run.py

```diff
@@ -181,3 +182,20 @@ def test_package_determination(
 
     assert "Cannot determine package name" not in caplog.text
     assert f"Determined package name: {package}" in caplog.text
+
+
+@mock.patch("os.execvpe", new=execvpe_mock)
```

Comment:
This would make the tests work very weirdly on Windows; maybe we should extract the actual execute-exit part to a standalone function and mock that instead.

Reply:
Looks like auto reinstall won't work on Windows. Maybe it's because we use a different way to run apps on Windows ?https://github.com/pypa/pipx/blob/4c0f2892c8f932f345a379b84bffe816c7c7206c/src/pipx/util.py#L390-L400

Reply:
Or we could just [run `python --version` to check for it](https://github.com/pypa/pipx/issues/449#issuecomment-660791532), so all platforms can be supported.